### DISCUSSION TITLE
Reload on PDF change

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#2903] [Ideas]
Bit of a side request, but probably one that would be easy.
Perhaps many of us doing design work on some document export it to PDF, and we have the PDF open (sometimes in more than one place if Sumatra is opening them in non-tabbed format, as I've found useful for this). As well as keeping tabs on actual PDF appearance, it can allow you to have a document open in several places if the original program doesn't allow it or is sluggish by comparison to a PDF.
Normally that then involves closing the PDF and reopening it to get the latest copy, which has to be done not a few times.
It would be good if Sumatra could detect when a file changes and ask if a reload is wanted. Perhaps just do this for local drives when a document gets a focus within Sumatra itself having the focus. Other option would be a File - Reload, which there is space for too. Maybe an option somewhere to check for file changes and reload automatically without asking (unless some uncommitted changes within sumatra exist in which case ask).
Cheers, David

**Comment by GitHubRulesOK** [comment]
One feature of SumatraPDF is that it refreshes while PDFs are in an editor like Edge or TeX or even notepad :-) (Not recommended as notepad can only handle a certain type of pdf, specially constructed like QPDF)

There are limits, older versions would need to be under 10 MB (current threshold is 32 MB) 

It is generally only PDF that works for auto reloads, and the file should be on local modern NTFS drive, to get MS change trigger, variable results on USB (AVOID sync or cloud folders)

Manual Refresh is `R` but no guarantees.

**Comment by meteorquake** [comment]
Thanks very much - R seems to do it in Sumatra, I had tried the usual Ctrl-R with no luck (I dont have PDFs view in browsers). It may have been a file or pagination size thing as I did a test with a small file and it automatically updated when I exported a PDF change but the one I had been working with didn't do so, it was a larger longer file. d

**Reply by GitHubRulesOK** [reply]
yes watch that 10/32 MB limit, it is caused by common needing to lock pdf files against changes, AFAIK Acrobat needs close & reopen or other fiddling but Foxit AFAIK does not have any size limit on PDF locking.

**Comment by bergsma-tim** [comment]
Thanks for this great pdf reader!  I make pdf in the RStudio environment.  (https://posit.co/downloads/).  Years ago, RStudio started bundling SumatraPDF with the installation.  Clicking on a pdf name in the files tab of the IDE caused it to be opened in SumatraPDF.  The _very best part_ about all this was that when the file was re-created (as for comments above) it was automatically reloaded in SumatraPDF.

Currently I have the latest versions of R/RStudio and SumatraPDF (installed by myself) running under windows 11.  I manually set the default pdf reader to SumatraPDF.  When I click on a pdf file name in the files tab of RStudio, it opens in SumatraPDF.  If I re-create the pdf and click again, I still see the stale version in SumatraPDF (but it can be manually reloaded with `R`).  If I open the file directly from the file menu of SumatraPDF, the RStudio environment refuses to re-create it. ```ERROR: The process cannot access the file because it is being used by another process. (os error 32)``` .  File size is 120.5 kb.

Is there something I can do to recover the auto-reload experience?

**Reply by GitHubRulesOK** [reply]
@bergsma-tim
Long time since using R but it has its own way of working independent of windows shell (thus possibly has its own file locking?)

When SumatraPDF is called from an application such as Windows Command it will for **PDF only** try to load that in memory without locking the file certainly when under 32 MB. Thus Latex editors can recompile most normal size PDF and the change in the windows file system will  trigger a refresh in SumatraPDF RUN TIME watching the file.

It is unusal to see another application say the file is locked by SumatraPDF. So for example I can edit this 30 MB PDF to say produced by ME no problem whilst viewing. I can also do similar with larger files too so the 32MB is not always a blocker the second example I am watching the change to a hyperlink in a 100MB file and on hover the change was both saved and auto refreshed, but both applications do not lock files on opening them.
![image](https://github.com/user-attachments/assets/62fe8431-4a3a-4611-8400-c91912947649)

102MB no problem the second image shows that again there is no block on returning the edit back to Reader ;-) whist watching on the left in sumatraPDF.
![image](https://github.com/user-attachments/assets/cc8e6388-094f-4292-bbec-27b8ac8de306)
![image](https://github.com/user-attachments/assets/39023afd-f0f8-4352-9351-a1e8d6bdd0f0)
